test/mlir/test/gc/gpu-runner/XeGPU/f16_mlp_32x4096x4096x4096.mlir

```diff
@@ -0,0 +1,77 @@
+// RUN: | gc-cpu-runner -e main --entry-point-result=void \
+// RUN:   --shared-libs=%mlir_runner_utils,%mlir_c_runner_utils,%opencl_runtime | FileCheck %s
+
+module {
+  func.func @linalg_mlp(%arg0: tensor<32x4096xf16>, %arg1: tensor<4096x4096xf16>, %arg2 : tensor<32x4096xf16>, 
+                        %arg3: tensor<4096x4096xf16>, %arg4 : tensor<32x4096xf16>) -> tensor<32x4096xf16> {
+    %cst = arith.constant 0.000000e+00 : f16
+    %0 = tensor.empty() : tensor<32x4096xf16>
+    %1 = linalg.fill ins(%cst : f16) outs(%0 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %2 = linalg.matmul ins(%arg0, %arg1 : tensor<32x4096xf16>, tensor<4096x4096xf16>)
+                       outs(%1 : tensor<32x4096xf16>) -> (tensor<32x4096xf16>)
+    %3 = tensor.empty() : tensor<32x4096xf16>
+    %4 = linalg.add ins(%arg2, %2 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                    outs(%3 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %5 = arith.constant dense<0.000000e+00> : tensor<32x4096xf16>
+    %6 = tensor.empty() : tensor<32x4096xf16>
+    %7 = linalg.max ins(%5, %4 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                    outs(%6 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+
+    %8 = tensor.empty() : tensor<32x4096xf16>
+    %9 = linalg.fill ins(%cst : f16) outs(%0 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %10 = linalg.matmul ins(%7, %arg3 : tensor<32x4096xf16>, tensor<4096x4096xf16>)
+                        outs(%9 : tensor<32x4096xf16>) -> (tensor<32x4096xf16>)
+    %11 = tensor.empty() : tensor<32x4096xf16>
+    %12 = linalg.add ins(%arg4, %10 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                     outs(%11 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+    %13 = arith.constant dense<0.000000e+00> : tensor<32x4096xf16>
+    %14 = tensor.empty() : tensor<32x4096xf16>
+    %15 = linalg.max ins(%13, %12 : tensor<32x4096xf16>, tensor<32x4096xf16>) 
+                     outs(%14 : tensor<32x4096xf16>) -> tensor<32x4096xf16>
+
+    return %15 : tensor<32x4096xf16>
```

Comment:
`insert-gpu-allocs` pass from IMEX seems to go crazy if we allocate & return a buffer in a function that contains `gpu.launch` (`linalg_mlp` in our case). For me it crashes the program at the end. It works fine though if we pass the final output buffer to `linalg_mlp`, e.g
```mlir
func.func @main() {
    %out = tensor.empty() : <...>

    func.call @linalg_mlp(..., %out)
}
```

If you haven't figured out a fix for this yet, we may simply stick to the 'pass output buffer to the func' model as it should be enough for our testing. We won't use `insert-gpu-allocs` in our final pipeline anyway.

Reply:
Thanks a lot! I encounter this problem and haven't figured out why. I will try to use your suggestion.